src/HexagonOptimize.cpp

```diff
@@ -91,6 +91,7 @@ struct Pattern {
         SwapOperands = 1 << 2,  ///< Swap operands prior to substitution.
         LosslessCastOp0 = 1 << 3, // Proceed only if lossless_cast on the first operand succeeds with the type of the first operand.
         LosslessCastOp1 = 1 << 4,  // Proceed only if lossless_cast on second operand succeeds with the type of the first operand.
+        IsOp1PowerOfTwo = 1 << 5, // Proceed only if the second operand, i.e. op1 is scalar constant which is a power of two.
```

Comment:
Seems like this flag should be named "ExactLog2Op1" or something like that, since it takes the log2 of the operand.